src/components/tools/polygon/PolygonTool.vue

```diff
@@ -154,11 +169,13 @@ export default defineComponent({
     });
 
     const onToolPlaced = () => {
-      if (currentImageID.value) {
-        placingToolID.value = activeToolStore.addTool({
-          imageID: currentImageID.value,
-          placing: true,
-        });
+      const imageID = currentImageID.value;
+      if (imageID) {
+        useHistoryStore().pushOperation(
+          { datasetID: imageID },
+          addToolOperation!
+        );
```

Comment:
What do you think about hiding `pushOperation` and its ilk in the tool stores?

Reply:
It might be good to do it later down the road. I'm opting not to for now because I don't want to make the tool stores undo/redo aware just yet.